File: tokens/platforms/swift/Sources/LaneShadowTheme/DomainColors.swift
import SwiftUI

public struct DomainColors: Sendable {
    public let waypointOnRoute: ColorSet
    public let waypointOffRoute: ColorSet
    public let waypointMixed: ColorSet
    public let enrichmentFast: ColorSet
    public let enrichmentExtended: ColorSet
    public let enrichmentCached: ColorSet
    public let deviationOriginalRoute: ColorSet
    public let deviationDetourPath: ColorSet
    public let deviationReconnectPoint: ColorSet
    public let locationPoiFill: ColorSet
    public let locationPoiRing: ColorSet
    public let locationPoiMuted: ColorSet
    public let locationPoiBg: ColorSet
    public let orange: ColorSet
}

extension DomainColors {
    static func build(from tokens: SemanticTokens) -> DomainColors {
        let L = tokens.color.light
        let D = tokens.color.dark
        return DomainColors(
            waypointOnRoute: makeColorSet(L, D, "waypointOnRoute"),
            waypointOffRoute: makeColorSet(L, D, "waypointOffRoute"),
            waypointMixed: makeColorSet(L, D, "waypointMixed"),
            enrichmentFast: makeColorSet(L, D, "enrichmentFast"),
            enrichmentExtended: makeColorSet(L, D, "enrichmentExtended"),
            enrichmentCached: makeColorSet(L, D, "enrichmentCached"),
            deviationOriginalRoute: makeColorSet(L, D, "deviationOriginalRoute"),
            deviationDetourPath: makeColorSet(L, D, "deviationDetourPath"),
            deviationReconnectPoint: makeColorSet(L, D, "deviationReconnectPoint"),
            locationPoiFill: makeColorSet(L, D, "locationPoiFill"),
            locationPoiRing: makeColorSet(L, D, "locationPoiRing"),
            locationPoiMuted: makeColorSet(L, D, "locationPoiMuted"),
            locationPoiBg: makeColorSet(L, D, "locationPoiBg"),
            orange: makeColorSet(L, D, "orange")
        )
    }
}

File: tokens/platforms/kotlin/src/main/kotlin/com/laneshadow/theme/DomainColors.kt
package com.laneshadow.theme

data class DomainColors(
    val waypointOnRoute: ColorSet,
    val waypointOffRoute: ColorSet,
    val waypointMixed: ColorSet,
    val enrichmentFast: ColorSet,
    val enrichmentExtended: ColorSet,
    val enrichmentCached: ColorSet,
    val deviationOriginalRoute: ColorSet,
    val deviationDetourPath: ColorSet,
    val deviationReconnectPoint: ColorSet,
    val locationPoiFill: ColorSet,
    val locationPoiRing: ColorSet,
    val locationPoiMuted: ColorSet,
    val locationPoiBg: ColorSet,
    val orange: ColorSet,
) {
    companion object {
        fun from(tokens: SemanticTokens, darkMode: Boolean): DomainColors {
            val m = if (darkMode) tokens.color.dark else tokens.color.light
            return DomainColors(
                waypointOnRoute = colorSet(m, "waypointOnRoute"),
                waypointOffRoute = colorSet(m, "waypointOffRoute"),
                waypointMixed = colorSet(m, "waypointMixed"),
                enrichmentFast = colorSet(m, "enrichmentFast"),
                enrichmentExtended = colorSet(m, "enrichmentExtended"),
                enrichmentCached = colorSet(m, "enrichmentCached"),
                deviationOriginalRoute = colorSet(m, "deviationOriginalRoute"),
                deviationDetourPath = colorSet(m, "deviationDetourPath"),
                deviationReconnectPoint = colorSet(m, "deviationReconnectPoint"),
                locationPoiFill = colorSet(m, "locationPoiFill"),
                locationPoiRing = colorSet(m, "locationPoiRing"),
                locationPoiMuted = colorSet(m, "locationPoiMuted"),
                locationPoiBg = colorSet(m, "locationPoiBg"),
                orange = colorSet(m, "orange"),
            )
        }
    }
}
